src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -90,6 +90,11 @@ class MethodBuilder(val fb: FunctionBuilder[_], mname: String, parameterTypeInfo
   def newLocal[T](name: String = null)(implicit tti: TypeInfo[T]): LocalRef[T] =
     new LocalRef[T](allocLocal[T](name))
 
+  def newField[T](implicit tti: TypeInfo[T]): ClassFieldRef[T] = newField()
+
+  def newField[T](name: String = null)(implicit tti: TypeInfo[T]): ClassFieldRef[T] =
```

Comment:
does it make sense to implement this as `fb.newField[T](name)`? (if yes: we can make `TypeInfo` a context bound: `[T: TypeInfo]` since we don't use the `tti` directly).